### PR TITLE
add implement some functions in string.go such as stringMSet stringMGet stringGetSet stringExist(#188)

### PR DESCRIPTION
**Pull request by wangchenguang123**
use in cli
a37e056a600dcd08cc121d28a39b70d

### CODE REVIEW

**Comment by sjcsjc123** [comment]
Good job! I will check it quickly!

File: cmd/client/string.go [hunk 1]
```diff
@@ -59,3 +59,73 @@ func stringDeleteKey(c *grumble.Context) error {
...
+
+// GetSet sets the value of a key and returns its old value
+func stringGetSet(c *grumble.Context) error {
+	key := c.Args.String("key")
+	value := c.Args.String("value")
+	if key == "" || value == "" {
+		fmt.Println("key or value is empty")
+		return nil
+	}
+	oldvalue, err := newClient().Get(key)
```
Comment (sjcsjc123) [inline cmd/client/string.go:71]: You can check this [pr](https://github.com/ByteStorage/FlyDB/pull/205). What you need to call is not this method, but rather adding a new method to the `lib/proto/dbs/db.proto` file. Then, use the `protoc` command to generate two files `lib/proto/dbs/db.pb.go` and `lib/proto/dbs/db_grpc.pb.go`, and supplement the remaining code in the `engine/grpc/service` package. You can pull the latest master branch, and you can see that there are any interfaces that you can try to implement, which needs to be combined with the methods under the string structure. The final step is to improve the methods under `engine/grpc/client/client.go` and then call them in `newClient()`

Comment (wangchenguang123) [reply cmd/client/string.go:71]: Thank you very much for your explanation. I am trying to understand your answer. I would like to ask, currently, I only need to write the methods in the engine/grpc/service/string.go file, as shown in the following image。
图片

Comment (qishenonly) [reply cmd/client/string.go:71]: > Thank you very much for your explanation. I am trying to understand your answer. I would like to ask, currently, I only need to write the methods in the engine/grpc/service/string.go file, as shown in the following image。 图片

Of course not, `engine/service/string.go` and `engine/client`, along with the corresponding `cli part`, need to be completed.

The `cli` is a method to invoke the `client`, and the `client` sends a request to the `service`.

Comment (wangchenguang123) [reply cmd/client/string.go:71]: Ok, thank you for your answer. I'll take a look again

Comment (wangchenguang123) [reply cmd/client/string.go:71]: > `engine/service/string.go`

i understand it thanks a lot. and i will try to finsh some functions in it.

**Pull request closed without merging**